applications/StructuralMechanicsApplication/python_scripts/structural_response_global_finite_differencing.py

```diff
@@ -0,0 +1,124 @@
+            nodal_variable = self.response_settings["nodal_sensitivity_variables"][i].GetString()
+            if nodal_variable == "SHAPE":
+                self._CalculateShapeSensitivities()
+            else:
+                raise NotImplementedError("FD for {} nodal_sensitivity_variables not implemented!".format(nodal_variable.GetString()) )
+
+        # element_sensitivity_variables
+        for i in range(self.response_settings["element_sensitivity_variables"].size()):
+            element_variable = self.response_settings["element_sensitivity_variables"][i].GetString()
+            raise NotImplementedError("FD for element_sensitivity_variables '{}' not implemented!".format(element_variable))
+
+        # condition_sensitivity_variables
+        for i in range(self.response_settings["condition_sensitivity_variables"].size()):
+            condition_variable = self.response_settings["condition_sensitivity_variables"][i].GetString()
+            raise NotImplementedError("FD for condition_sensitivity_variables '{}' not implemented!".format(condition_variable))
+
+    def _CalculateShapeSensitivities(self):
+        if self.response_settings["sensitivity_model_part_name"].GetString() == self.model_part.Name:
+            sensitivity_model_part = self.model_part
+        elif self.response_settings["sensitivity_model_part_name"].GetString() == "":
+            sensitivity_model_part = self.model_part
+        else:
+            sensitivity_model_part = self.model_part.GetSubModelPart(self.response_settings["sensitivity_model_part_name"].GetString())
+        perturbation = self.response_settings["step_size"].GetDouble()
+        unperturbed_value = self.GetValue()
+
+        for node in sensitivity_model_part.Nodes:
+            for direction in range(3):
+                perturbed_model = KratosMultiphysics.Model()
+                identifier = "Node_{}_direction_{}".format(node.Id, direction)
+                perturbed_response = self._CreateNewTmpResponse(identifier, perturbed_model)
+                perturbed_response.Initialize()
+                # perturb
+                model_part = perturbed_response.primal_model_part #TODO find a better way to get the model part
+                tmp_node = model_part.Nodes[node.Id]
+                if direction == 0:
+                    tmp_node.X += perturbation
+                    tmp_node.X0 += perturbation
+                if direction == 1:
+                    tmp_node.Y += perturbation
+                    tmp_node.Y0 += perturbation
+                if direction == 2:
+                    tmp_node.Z += perturbation
+                    tmp_node.Z0 += perturbation
+
+                perturbed_response.InitializeSolutionStep()
+                perturbed_response.CalculateValue()
+                perturbed_value = perturbed_response.GetValue()
+
+                sensitivity = node.GetSolutionStepValue(KratosMultiphysics.SHAPE_SENSITIVITY)
```

Comment:
There will be a problem if one of the adjoint response functions is used, because in this case the variable ```SHAPE_SENSITIVITY``` is added to the model part which belongs to the adjoint model.

Reply:
fixed it